File: test_go/src/pattern/template_method/barista/coffee.go
package simple_barista

import "fmt"

type Coffee struct {
	baseCaffeineBeverage
}

func NewCoffee() CaffeineBeverage {
	this := new(Coffee)
	this.baseCaffeineBeverage.CaffeineBeverage = this
	return this
}

func (this *Coffee) Brew() {
	fmt.Println("Dripping Coffee through filter")
}

func (this *Coffee) AddCondiments() {
	fmt.Println("Adding Sugar and Milk")
}

File: java_test/src/DesignPattern/TemplateMethod/Barista/Coffee.java
package DesignPattern.TemplateMethod.Barista;

/**
 * Created by Administrator on 2018/4/28.
 */
public class Coffee extends CaffeineBeverage {
    public void brew() {
        System.out.println("Dripping Coffee through filter");
    }

    public void addCondiments() {
        System.out.println("Adding Sugar and Milk");
    }
}
